### ISSUE TITLE
Handler() doesn't wrap http.ResponseWriter fully

### ISSUE THREAD

**Issue by muirrn** [aws/aws-xray-sdk-go#80]
`xray.Handler()` leaves the `http.ResponseWriter` no longer implementing `http.Flusher`, `http.Hijacker` and `http.CloseNotifier`. See https://github.com/gorilla/handlers/blob/7e6a874cdc0efb71a260254cc0e30d5415b6a5bb/logging.go#L58 for an example of how you can maintain those interfaces when wrapping the response writer.

Here is a simple program to demonstrate the issue:

```
package main

import (
	"fmt"
	"net/http"
	"net/http/httptest"

	"github.com/aws/aws-xray-sdk-go/xray"
)

func main() {
	mux := http.NewServeMux()

	mux.Handle("/xray", xray.Handler(xray.NewFixedSegmentNamer("test"), http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		_, isFlusher := w.(http.Flusher)
		_, isHijacker := w.(http.Hijacker)
		fmt.Printf("xray: flusher=%t hijacker=%t\n", isFlusher, isHijacker)
	})))

	mux.Handle("/no_xray", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		_, isFlusher := w.(http.Flusher)
		_, isHijacker := w.(http.Hijacker)
		fmt.Printf("xray: flusher=%t hijacker=%t\n", isFlusher, isHijacker)
	}))

	server := httptest.NewServer(mux)

	http.Get(server.URL + "/xray")
	http.Get(server.URL + "/no_xray")
}
```

**Comment by luluzhao** [comment]
Hello @muirrn , thank you for reaching out to us. Based on the example you attached, I found out the difference that `xray.Handler` changed the behavior of `http.ResponseWriter`. We will add this into our backlog and make sure it can be included in our following releases. We are also welcome to accept PR regarding for this one. Feel free to submit it if you have and we'd love to prioritize for reviewing it. Thanks.

**Comment by chrisradek** [comment]
Thanks for opening this issue!

The fix for this is included in [V1.0.0-rc.11](https://github.com/aws/aws-xray-sdk-go/releases/tag/V1.0.0-rc.11).
